### PR TITLE
Core: Rewrite start inventory from pool code

### PR DESCRIPTION
**Pull request by NewSoupVi**
I think the old code is pretty ugly and unpythonic

Here's my proposal for a better version

Even had some room to spare for comments, since I made it a couple lines shorter

### CODE REVIEW

File: Main.py [hunk 1]
```diff
@@ -149,45 +149,45 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
...
-                                    f" is trying to remove items from their pool that don't exist: {remaining_items}")
-                    # find all filler we generated for the current player and remove until it matches 
-                    removables = [item for item in new_items if item.player == player]
-                    for _ in range(sum(remaining_items.values())):
-                        new_items.remove(removables.pop())
-        assert len(multiworld.itempool) == len(new_items + old_items), "Item Pool amounts should not change."
-        multiworld.itempool[:] = new_items + old_items
+                new_itempool.append(item)
+
+        unfound_items_per_player = {
```
Comment (Exempt-Medic) [inline Main.py:173]: I feel like this section could all be inside an `if overall_target:` and then at the end have:
```py
else:
    for player, target in target_per_player.items():
        new_itempool += [multiworld.worlds[player].create_filler() for _ in range(target_per_player[player])]
```

**Review by Exempt-Medic — APPROVED** [review]
One comment, otherwise the changes LGTM. Tested various configurations of start_inventory_from_pool and all behavior was identical to how it was before.

File: Main.py (hunk 1 above)
Comment (NewSoupVi) [reply Main.py:173]: I do think this makes the code slightly more complicated, at probably little performance gain (The full iteration of the multiworld itempool is gonna be so much slower than anything that happens after it)

But I see your point. I'll think about it. I don't like the idea of just wrapping it in an extra indent, but maybe there is a different cute solution here.

Comment (NewSoupVi) [reply Main.py:173]: I timed it roughly, this change barely has any impact.

Comment (Exempt-Medic) [reply Main.py:173]: Yeah, I went and timed it myself with 10K players and it saved 0.0054 seconds on average over 20 trials (the range was 0.005-0.006) so... yeah, not worth

Comment (NewSoupVi) [reply Main.py]: I made a different improvement that should make this better in the worst cases

**Comment by NewSoupVi** [comment]
I have now opted to remove `overall_target`.

I understand why it exists, it does make the average case about 20% faster, which is probably worth it for a full iteration of the multiworld itempool.

However, it 1. makes the code more confusing and complicated, and 2. **Now that we have this warn behavior**, I imagine in big multiworlds, statistically, one player will make use of it, and as soon as **one** item can't be found, we're in the bad code path anyway.

**Review by qwint — APPROVED** [review]
code looks correct and cuts some corners that should improve performance but didn't actually run the code yet to see if i can catch any issues (medic seems to have ran through most of them tho lol)

**Pull request merged**